### PR TITLE
Tracing: Docs and config improvements for Tempo/Jaeger/Zipkin

### LINKED ISSUE #64659: Trace to log custom query not working in provisioning file
**What happened**:

I am configuring the data source in grafana using a provisioning file and everything works except for custom query in the "trace to log" section

**What you expected to happen**:

the use custom query should be toggled on and the query populated

**How to reproduce it (as minimally and precisely as possible)**:

```
datasources:
- name: Tempo
  type: tempo
  uid: tempo
  access: proxy
  orgId: 1
  url: http://tempo:3200
  basicAuth: false
  isDefault: false
  editable: true
  jsonData:
    httpMethod: GET
    tracesToLogs:
      datasourceUid: loki
      spanStartTimeShift: '-1h'
      spanEndTimeShift: '+1h'
      filterByTraceID: false
      filterBySpanID: false
      customQuery: true
      query: '{{exporter="OTLP"} | json | line_format "{{.body}}" | json | trace_id="${__span.traceId}" | span_id="${__span.spanId}"}'
    serviceMap:
      datasourceUid: prometheus
```
image

**Anything else we need to know?**:

if i manually toggle it on and enter the query, everything works fine :) but i want to use the provisioning file since I am using docker containers

**Environment**:
- Grafana version: 

v9.3.3

image

- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by joey-grafana**
**What is this feature?**

Improves the docs for Tempo, Jaeger & Zipkin. 

**Why do we need this feature?**

Makes how to use and configure these data sources clearer. Updates provisioning examples, terminology and descriptive text.

**Who is this feature for?**

Anyone who uses the Tempo, Jaeger or Zipkin data sources.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/64659

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [x] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [x] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

File: docs/sources/datasources/jaeger/_index.md [hunk 1]
```diff
@@ -153,41 +152,55 @@ You can choose one of three options:
...
+        spanStartTimeShift: '1h'
+        spanEndTimeShift: '-1h'
         tags: ['job', 'instance', 'pod', 'namespace']
-        mappedTags: [{ key: 'service.name', value: 'service' }]
-        mapTagNamesEnabled: false
+        filterByTraceID: false
+        filterBySpanID: false
+        customQuery: true
+        query: 'method="${__span.tags.method}"'
+      # Note: tracesToLogs used before Grafana version 9.4
```
Comment (aocenas) [inline docs/sources/datasources/jaeger/_index.md:182]: Should we keep this? I mean if somebody uses <9.4 Grafana they should look at that version of docs so it's not lost, but I guess people should not use this anymore.

Comment (joey-grafana) [reply docs/sources/datasources/jaeger/_index.md]: No we shouldn't. Older documentation versions will suffice. Thank you :)

**Review by aocenas — APPROVED** [review]

File: docs/sources/datasources/jaeger/_index.md [hunk 2]
```diff
@@ -95,23 +94,23 @@ To use a variable you need to wrap it in `${}`. For example: `${__span.name}`.
...
+| **Data source**           | Defines the target data source. You can select only Loki or Splunk \[logs\] data sources.                                                                                                                                                                                  |
+| **Span start time shift** | Shifts the start time for the logs query, based on the span's start time. You can use time units, such as `5s`, `1m`, `3h`. To extend the time to the past, use a negative value. Default: `0`.                                                                            |
+| **Span end time shift**   | Shifts the end time for the logs query, based on the span's end time. You can use time units. Default: `0`.                                                                                                                                                                |
+| **Tags**                  | Defines the the tags to use in the logs query. Default: `cluster`, `hostname`, `namespace`, `pod`. You can change the tag name for example to remove dots from the name if they are not allowed in the target data source. For example map `http.status` to `http_status`. |
```
Comment (knylander-grafana) [inline docs/sources/datasources/jaeger/_index.md:102]:
```suggestion
| **Tags**                  | Defines the the tags to use in the logs query. Default: `cluster`, `hostname`, `namespace`, `pod`. You can change the tag name for example to remove dots from the name if they are not allowed in the target data source. For example: map `http.status` to `http_status`. |
```

File: docs/sources/datasources/jaeger/_index.md [hunk 3]
```diff
@@ -283,8 +286,14 @@ If the file has multiple traces, Grafana visualizes its first trace.
 }
 ```
 
-## Link a trace ID from logs
+## Link to a trace ID from logs
+
+You can link to Jaeger traces from logs in Loki, Elasticsearch, Splunk and other logs data sources by configuring an internal link.
```
Comment (knylander-grafana) [inline docs/sources/datasources/jaeger/_index.md:291]: Using a serial comma helps keep things clearer for readers. 

```suggestion
You can link to Jaeger traces from logs in Loki, Elasticsearch, Splunk, and other logs data sources by configuring an internal link.
```

File: docs/sources/datasources/jaeger/_index.md [hunk 4]
```diff
@@ -283,8 +286,14 @@ If the file has multiple traces, Grafana visualizes its first trace.
...
+
+To configure this feature, see the [Derived fields]({{< relref "../loki#configure-derived-fields" >}}) section of the Loki data source docs, or the [Data links]({{< relref "../elasticsearch#data-links" >}}) section of the Elasticsearch or Splunk data source docs.
```
Comment (knylander-grafana) [inline docs/sources/datasources/jaeger/_index.md:293]:
```suggestion
To configure this feature, see the [Derived fields]({{< relref "../loki#configure-derived-fields" >}}) section of the Loki data source docs or the [Data links]({{< relref "../elasticsearch#data-links" >}}) section of the Elasticsearch or Splunk data source docs.
```

File: docs/sources/datasources/tempo/_index.md [hunk 5]
```diff
@@ -331,8 +350,14 @@ To open a query in Prometheus with the span name of that row automatically set i
 
 To open a query in Tempo with the span name of that row automatically set in the query, click a row in the **links** column.
 
-## Link a trace ID from logs
+## Link to a trace ID from logs
+
+You can link to Tempo traces from logs in Loki, Elasticsearch, Splunk and other logs data sources by configuring an internal link.
+
+To configure this feature, see the [Derived fields]({{< relref "../loki#configure-derived-fields" >}}) section of the Loki data source docs, or the [Data links]({{< relref "../elasticsearch#data-links" >}}) section of the Elasticsearch or Splunk data source docs.
```
Comment (knylander-grafana) [inline docs/sources/datasources/tempo/_index.md:357]:
```suggestion
To configure this feature, see the [Derived fields]({{< relref "../loki#configure-derived-fields" >}}) section of the Loki data source docs or the [Data links]({{< relref "../elasticsearch#data-links" >}}) section of the Elasticsearch or Splunk data source docs.
```

File: docs/sources/datasources/zipkin/_index.md [hunk 6]
```diff
@@ -205,8 +255,14 @@ If the file has multiple traces, Grafana visualizes its first trace.
 ]
 ```
 
-## Link a trace ID from logs
+## Link to a trace ID from logs
+
+You can link to Zipkin traces from logs in Loki, Elasticsearch, Splunk and other logs data sources by configuring an internal link.
+
+To configure this feature, see the [Derived fields]({{< relref "../loki#configure-derived-fields" >}}) section of the Loki data source docs, or the [Data links]({{< relref "../elasticsearch#data-links" >}}) section of the Elasticsearch or Splunk data source docs.
```
Comment (knylander-grafana) [inline docs/sources/datasources/zipkin/_index.md:262]:
```suggestion
To configure this feature, see the [Derived fields]({{< relref "../loki#configure-derived-fields" >}}) section of the Loki data source docs or the [Data links]({{< relref "../elasticsearch#data-links" >}}) section of the Elasticsearch or Splunk data source docs.
```

File: docs/sources/explore/trace-integration.md [hunk 7]
```diff
@@ -73,18 +73,24 @@ You can optionally expand the node graph for the displayed trace. Depending on t
...
 
 {{< figure src="/static/img/docs/explore/trace-to-log-7-4.png" class="docs-image--no-shadow" max-width= "600px"  caption="Screenshot of the trace view in Explore with icon next to the spans" >}}
 
 Click the document icon to open a split view in Explore with the configured data source and query relevant logs for the span.
 
+### Trace to metrics
+
+> **Note:** This feature is currently in beta & behind the `traceToMetrics` feature toggle.
+
+You can navigate from a span in a trace view directly to metrics relevant for that span. This is available for Tempo, Jaeger, and Zipkin data sources at this moment. Refer to their [relevant documentation](https://grafana.com/docs/grafana/latest/datasources/tempo/#trace-to-metrics) for instructions on how to configure this feature.
```
Comment (knylander-grafana) [inline docs/sources/explore/trace-integration.md:86]:
```suggestion
You can navigate from a span in a trace view directly to metrics relevant for that span. This feature is available for Tempo, Jaeger, and Zipkin data sources. Refer to their [relevant documentation](https://grafana.com/docs/grafana/latest/datasources/tempo/#trace-to-metrics) for configuration instructions.
```

File: docs/sources/explore/trace-integration.md [hunk 8]
```diff
@@ -73,18 +73,24 @@ You can optionally expand the node graph for the displayed trace. Depending on t
...
+
 ## Service Graph view
 
 The Service Graph view visualizes the span metrics (traces data for rates, error rates, and durations (RED)) and service graphs.
 Once the requirements are set up, this pre-configured view is immediately available.
 
-For more information, refer to the [Service Graph view table section](https://grafana.com/docs/grafana/latest/datasources/tempo/#service-graph-view-table) of the Tempo data source page and the [service graph view page](https://grafana.com/docs/tempo/latest/metrics-generator/service-graph-view/) in the Tempo documentation.
+For more information, refer to the [Service Graph view section](https://grafana.com/docs/grafana/latest/datasources/tempo/#open-the-service-graph-view) of the Tempo data source page and the [service graph view page](https://grafana.com/docs/tempo/latest/metrics-generator/service-graph-view/) in the Tempo documentation.
```
Comment (knylander-grafana) [inline docs/sources/explore/trace-integration.md:93]:
```suggestion
For more information, refer to the [Service Graph view section]({{< relref "/docs/grafana/latest/datasources/tempo/#open-the-service-graph-view" >}}) of the Tempo data source page and the [service graph view page]({{< relref "/docs/tempo/latest/metrics-generator/service-graph-view/" >}}) in the Tempo documentation.
```

**Review by knylander-grafana — APPROVED** [review]
Thank you for adding documentation! Approving the changes, with some minor suggestions.

**Review by chri2547 — APPROVED** [review]
@joey-grafana , after you've accepted @knylander-grafana suggestions, hit me up directly to merge this for you (if you are unable to merge yourself). Thanks for the great work!

**Pull request merged**